sale_order_import/wizard/sale_order_import.py

```diff
@@ -0,0 +1,663 @@
+# Copyright 2016-2017 Akretion
+# @author: Alexis de Lattre <alexis.delattre@akretion.com>
+# Copyright 2022 Camptocamp
+# @author: Simone Orsi <simahawk@gmail.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import logging
+import mimetypes
+from base64 import b64decode, b64encode
+
+from lxml import etree
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+from odoo.osv.expression import AND
+from odoo.tools import config, float_compare, float_is_zero
+
+logger = logging.getLogger(__name__)
+
+
+class SaleOrderImport(models.TransientModel):
+    _name = "sale.order.import"
+    _description = "Sale Order Import from Files"
+
+    state = fields.Selection(
+        [("import", "Import"), ("update", "Update")], default="import"
+    )
+    partner_id = fields.Many2one("res.partner", string="Customer")
+    csv_import = fields.Boolean(default=False, readonly=True)
```

Comment:
This flag should be removed and a new selection field `import_type` should be added.
`supported_types` below should depend on this and a `_parse_{import_type}` should be used by `_parse_file`